From 3b5954b321963b7cee91c78ba8f57ba687d0aae6 Mon Sep 17 00:00:00 2001
From: David S. Wang <dsw@cloudera.com>
Date: Thu, 28 Mar 2013 13:53:03 -0700
Subject: [PATCH 07/96] CDH-7252: MR1 + HDFS 2.0 profile

Reason: Test
Ref: CDH-7252
Author: Matteo Bertozzi
(cherry picked from commit 2a187418a37cdc72a55bc72824afaa97823d3f9b)
---
 pom.xml |   99 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 files changed, 99 insertions(+), 0 deletions(-)

diff --git a/pom.xml b/pom.xml
index 9fa543f..ed71b4d 100644
--- a/pom.xml
+++ b/pom.xml
@@ -1967,6 +1967,105 @@
     </profile>
 
     <!--
+      profile for building against Hadoop 0.23.0 with MR1. Activate using:
+       mvn -Dhadoop.profile=mr1
+    -->
+    <profile>
+      <id>hadoop-mr1</id>
+      <activation>
+        <property>
+          <name>hadoop.profile</name>
+          <value>mr1</value>
+        </property>
+      </activation>
+      <properties>
+        <hadoop.version>${cdh.mr1.version}</hadoop.version>
+        <slf4j.version>1.6.1</slf4j.version>
+      </properties>
+      <dependencies>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-core</artifactId>
+          <version>${hadoop.version}</version>
+        </dependency>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-common</artifactId>
+          <version>${cdh.hadoop.version}</version>
+        </dependency>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-common</artifactId>
+          <version>${cdh.hadoop.version}</version>
+          <scope>compile</scope>
+          <type>test-jar</type>
+        </dependency>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-hdfs</artifactId>
+          <version>${cdh.hadoop.version}</version>
+        </dependency>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-hdfs</artifactId>
+          <version>${cdh.hadoop.version}</version>
+          <type>test-jar</type>
+        </dependency>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-test</artifactId>
+          <version>${hadoop.version}</version>
+        </dependency>
+      </dependencies>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>build-helper-maven-plugin</artifactId>
+            <executions>
+              <execution>
+                <id>add-test-resource</id>
+                <goals>
+                  <goal>add-test-resource</goal>
+                </goals>
+                <configuration>
+                  <resources>
+                    <resource>
+                      <directory>src/test/resources</directory>
+                      <includes>
+                        <include>hbase-site.xml</include>
+                      </includes>
+                    </resource>
+                  </resources>
+                </configuration>
+              </execution>
+            </executions>
+          </plugin>
+          <plugin>
+            <artifactId>maven-dependency-plugin</artifactId>
+            <executions>
+              <execution>
+                <id>create-mrapp-generated-classpath</id>
+                <phase>generate-test-resources</phase>
+                <goals>
+                  <goal>build-classpath</goal>
+                </goals>
+                <configuration>
+                  <!-- needed to run the unit test for DS to generate
+                  the required classpath that is required in the env
+                  of the launch container in the mini mr/yarn cluster
+                  -->
+                  <outputFile>${project.build.directory}/test-classes/mrapp-generated-classpath</outputFile>
+                </configuration>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+
+
+    <!--
       profile for building against Hadoop 0.23.0. Activate using:
        mvn -Dhadoop.profile=23
     -->
-- 
1.7.0.4

